include/nlohmann/detail/input/binary_reader.hpp

```diff
@@ -1665,9 +1663,9 @@ class binary_reader
     */
     std::string get_token_string() const
     {
-        std::stringstream ss;
-        ss << std::setw(2) << std::uppercase << std::setfill('0') << std::hex << current;
-        return ss.str();
+        char cr[3];
+        snprintf(cr, 3, "%.2X", current);
+        return std::string{cr};
```

Comment:
Except that C++11 made that undefined, and the non-const `data()` doesn't exist until C++17, so never mind.

Reply:
Even with non-const `.data()` it is undefined.
> [The pointer is such that the range [data(); data() + size()) is valid](http://en.cppreference.com/w/cpp/string/basic_string/data)

The main problem is the null-termination of `snprintf`, which always results in a string with an +1 increased stringlength compared to the the string created before.

Reply:
That's fine, `size()` is 2, so `cr[0]` to `cr[2]` is valid, and that's exactly what `snprintf` with a size of `3` will write to, including the nul terminator.  In any case, we can't use it, as we can only require C++11.

Reply:
Ah, I see. Scrolling down to the return value, the definition interval is closed 
> data() + i == &operator[](i) for every i in [0, size()]. | (since C++11)

while in the brief, the definition interval is only an open one which excludes [size()].